profitbricks/src/main/java/org/jclouds/profitbricks/domain/IpBlock.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.domain;
+
+import com.google.auto.value.AutoValue;
+import java.util.List;
+import org.jclouds.javax.annotation.Nullable;
+
+@AutoValue
+public abstract class IpBlock {
+
+   public abstract String id();
+
+   public abstract Location location();
+
+   public abstract List<PublicIp> publicIps();
+
+   @Nullable
+   public abstract List<String> ips();
+
+   public static IpBlock create(String id, Location location, List<PublicIp> publicIps, List<String> ips) {
+      return new AutoValue_IpBlock(id, location, publicIps, ips);
```

Comment:
To enforce the immutability of the list, if not null you should pass an `ImmutableList.copyOf(ips)` to the auto value constructor.